include/seqan3/alignment/matrix/detail/coordinate_matrix.hpp

```diff
@@ -162,21 +161,23 @@ class coordinate_matrix_iterator
  *
  * \details
  *
- * In the alignment algorithm an additional matrix is needed to represent the coordinates of the processed cells.
+ * This matrix emulates a two-dimensional index, such that each cell inside of the alignment matrix can be located
+ * through a unique coordinate (=index). In the alignment algorithm this matrix is paired with the alignment matrix
+ * to form an indexed alignment matrix.
+ *
  * This matrix is cheap as it stores only the dimensions of the matrix and does not allocate any memory for the
  * coordinates. It uses the seqan3::detail::matrix_coordinate_iterator to iterate
  * over the columns of this virtual matrix. When the seqan3::detail::matrix_coordinate_iterator is dereferenced it
  * returns an on-the-fly constructed range ([prvalue](https://en.cppreference.com/w/cpp/language/value_category))
- * representing the current coordinate column. The value type of this range is seqan3::detail::matrix_coordinate.
+ * representing the current indexed column. The reference type of this range is seqan3::detail::matrix_coordinate.
  */
 class coordinate_matrix
 {
 private:
-
-    //!\brief The column index marking the end of columns (size of one row).
-    column_index_type<size_t> m_end_column_index{};
-    //!\brief The row index marking the end of rows (size of one column).
-    row_index_type<size_t> m_end_row_index{};
+    //!\brief Represents the size of a row (number of columns).
+    size_t number_of_columns{};
+    //!\brief Represents the size of a column (number of rows).
+    size_t number_of_rows{};
```

Comment:
I think this would be an example for the name `rows_count`. Don't rename it; Just observation